src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -3862,19 +3841,35 @@ template<typename R, typename... Rx>
   }
 
 // SVE unpack vector elements
-#define INSN(NAME, op) \
-  void NAME(FloatRegister Zd, SIMD_RegVariant T, FloatRegister Zn) { \
-    starti;                                                          \
-    assert(T != B && T != Q, "invalid size");                        \
-    f(0b00000101, 31, 24), f(T, 23, 22), f(0b1100, 21, 18);          \
-    f(op, 17, 16), f(0b001110, 15, 10), rf(Zn, 5), rf(Zd, 0);        \
+protected:
+  void _sve_xunpk(bool is_unsigned, bool is_high, FloatRegister Zd, SIMD_RegVariant T, FloatRegister Zn) {
+    starti;
+    assert(T != B && T != Q, "invalid size");
+    f(0b00000101, 31, 24), f(T, 23, 22), f(0b1100, 21, 18);
+    f(is_unsigned ? 1 : 0, 17), f(is_high ? 1 : 0, 16),
+    f(0b001110, 15, 10), rf(Zn, 5), rf(Zd, 0);
   }
 
-  INSN(sve_uunpkhi, 0b11); // Signed unpack and extend half of vector - high half
-  INSN(sve_uunpklo, 0b10); // Signed unpack and extend half of vector - low half
-  INSN(sve_sunpkhi, 0b01); // Unsigned unpack and extend half of vector - high half
-  INSN(sve_sunpklo, 0b00); // Unsigned unpack and extend half of vector - low half
-#undef INSN
+public:
+  // Unsigned unpack and extend half of vector - high half
+  void sve_uunpkhi(FloatRegister Zd, SIMD_RegVariant T, FloatRegister Zn) {
+    _sve_xunpk(/* is_unsigned */ true, /* is_high */ true, Zd, T, Zn);
+  }
+
+  // Unsigned unpack and extend half of vector - low half
+  void sve_uunpklo(FloatRegister Zd, SIMD_RegVariant T, FloatRegister Zn) {
+    _sve_xunpk(/* is_unsigned */ true, /* is_high */ false, Zd, T, Zn);
+  }
+
+  // Signed unpack and extend half of vector - high half
+  void sve_sunpkhi(FloatRegister Zd, SIMD_RegVariant T, FloatRegister Zn) {
+    _sve_xunpk(/* is_unsigned */ false, /* is_high */ true, Zd, T, Zn);
+  }
+
+  // Signed unpack and extend half of vector - low half
+  void sve_sunpklo(FloatRegister Zd, SIMD_RegVariant T, FloatRegister Zn) {
+    _sve_xunpk(/* is_unsigned */ false, /* is_high */ false, Zd, T, Zn);
+  }
```

Comment:
This code expansion does not look right. You should be able to make this change without so much code expansion.

```
#define INSN(NAME, unsigned, high)
void name(FloatRegister Zd, SIMD_RegVariant T, FloatRegister Zn) { \
  _sve_xunpk(unsigned, high, T, Zn);                                            \
}
  INSN(sve_uunpkhi, true, true)
...
```